eng/automation/GenerateAndBuildLib.ps1

```diff
@@ -0,0 +1,149 @@
+        Write-Error $_
+    }
+    Write-Host "Cannot find swagger info"
+    exit 1
+}
+
+function Get-InputJsonFileInfo()
+{
+    param(
+        [string]$dir,
+        [string]$AUTOREST_CONFIG_FILE = "autorest.md"
+    )
+    Set-Location $dir
+    $inputfileInfoRegex = ".*github.*.com\/(?<org>.*)\/azure-rest-api-specs\/blob\/(?<commitID>[0-9a-f]{40})\/specification\/(?<specName>.*)\/data-plane\/?<filePath>"
+    $rawInputfileInfoRegex = ".*github.*.com\/(?<org>.*)\/azure-rest-api-specs\/(?<commitID>[0-9a-f]{40})\/specification\/(?<specName>.*)\/data-plane\/?<filePath>"
+    $inputfileNoCommitRegex = ".*github.*.com\/(?<org>.*)\/azure-rest-api-specs\/(blob\/)?(?<branch>.*)\/specification\/(?<specName>.*)\/data-plane\/?<filePath>"
+    try
+    {
+        $content = Get-Content .\$AUTOREST_CONFIG_FILE -Raw
+        if ($content -match $inputfileInfoRegex)
+        {
+            return $matches["org"], $matches["specName"], $matches["commitID"], $matches["filePath"]
+        }
+        if ($content -match $rawInputfileInfoRegex)
+        {
+            return $matches["org"], $matches["specName"], $matches["commitID"], $matches["filePath"]
+        }
+        if ($content -match $inputfileNoCommitRegex)
+        {
+            return $matches["org"], $matches["specName"], "", $matches["filePath"]
+        }
+    }
+    catch
+    {
+        Write-Error "Error parsing input file info"
+        Write-Error $_
+    }
+    Write-Host "Cannot find input file info"
+    exit 1
+}
+function New-DataPlanePackageFolder() {
+  param(
+      [string]$service,
+      [string]$namespace,
+      [string]$sdkPath = "",
+      [string]$inputfile = "",
+      [string]$securityScope,
+      [string]$AUTOREST_CONFIG_FILE = "autorest.md",
+      [string]$outputJsonFile = "output.json"
+  )
+
+  $sdkPath = $sdkPath -replace "\\", "/"
+
+  $projectFolder="$sdkPath/sdk/$service/$namespace"
+  if (Test-Path -Path $projectFolder) {
+    Write-Host "Path exists!"
+  } else {
+    Write-Host "Path doesn't exist. create template."
+    if ($inputfile -eq "") {
+        Write-Error "Error: input file should not be empty."
+        exit 1
+    }
+    dotnet new -i $sdkPath/eng/templates/Azure.ServiceTemplate.Template
+    $projectFolder="$sdkPath/sdk/$service/$namespace"
+    Write-Host "Create project folder $projectFolder"
+    New-Item -Path $projectFolder -ItemType Directory
+    Set-Location $projectFolder
+    $namespaceArray = $namespace.Split(".")
+    if ( $namespaceArray.Count -ne 3) {
+        Write-Error "Error: invalid namespace name."
+        exit 1
+    }
+
+    $libraryName = $namespaceArray[2]
+    dotnet new dataplane --libraryName $libraryName --swagger $inputfile --securityScopes $securityScope --includeCI true --force
+    dotnet sln remove src\$namespace.csproj
+    dotnet sln add src\$namespace.csproj
+    dotnet sln remove tests\$namespace.Tests.csproj
+    dotnet sln add tests\$namespace.Tests.csproj
```

Comment:
You should be able to do this with post action: https://github.com/dotnet/templating/blob/main/docs/Post-Action-Registry.md#add-projects-to-a-solution-file

Reply:
ok, I will try it.

Reply:
Not need to remove/add project. Now the projects can be successfully restored.